content/developers/dogstatsd/data_types.md

```diff
@@ -0,0 +1,309 @@
+---
+title: Data Types and Tags
+kind: documentation
+description: Overview of the features of DogStatsD, including data types and tagging.
+further_reading:
+- link: "developers/dogstatsd"
+  tag: "Documentation"
+  text: Introduction to DogStatsD
+- link: "developers/libraries"
+  tag: "Documentation"
+  text: Official and Community-contributed API and DogStatsD client libraries
+- link: "https://github.com/DataDog/dd-agent/blob/master/dogstatsd.py"
+  tag: "Github"
+  text: DogStatsD source code
+---
+
+While StatsD accepts only metrics, DogStatsD accepts all three of the major Datadog data types: metrics, events, and service checks. This section shows typical use cases for each type, and introduces tagging, which is specific to DogStatsD.
+
+Each example is in Python using the [official Datadog Python client][2], but each data type shown is supported similarly in [other DogStatsD client libraries][1].
+
+## Metrics
+
+Counters, gauges, and sets are familiar to StatsD users. Histograms are specific to DogStatsD. Timers, which exist in StatsD, are a sub-set of histograms in DogStatsD.
+
+### Counters
+
+Counters track how many times something happens _per second_, such as page views. In this example, we increment a metric called `web.page_views` each time our `render_page` function is called.
+
+For Python:
+```python
+
+def render_page():
+    """ Render a web page. """
+    statsd.increment('web.page_views')
+    return 'Hello World!'
+```
+
+For Ruby:
+```ruby
+def render_page()
+  # Render a web page.
+  statsd.increment('web.page_views')
+  return 'Hello World!'
+end
+```
+
+With this one line of code we can start graphing the data. Here's an example:
+
+{{< img src="developers/metrics/graph-guides-metrics-page-views.png" alt="graph guides metrics page views" responsive="true" popup="true">}}
+
+Note that StatsD counters are normalized over the flush interval to report per-second units. In the graph above, the marker is reporting 35.33 web page views per second at ~15:24. In contrast, if one person visited the web page each second, the graph would be a flat line at y = 1. To increment or measure values over time, see [gauges](#gauges).
+
+We can also count by arbitrary numbers. Suppose we wanted to count the number of bytes processed by a file uploading service. We increment a metric called `file_service.bytes_uploaded` by the size of the file each time our `upload_file` function is called:
+
+For Python:
+```python
+
+def upload_file(file):
+    statsd.increment('file_service.bytes_uploaded', file.size())
+    save_file(file)
+    return 'File uploaded!'
+```
+
+Note that for counters coming from another source that are ever-increasing and never reset -- for example, the number of queries from MySQL over time -- we track the rate between flushed values. While there currently isn't an elegant solution to get raw counts within Datadog, you may want to apply a function to
```

Comment:
`While there currently isn't an elegant solution`, should we be that partial in the dev section?